libetrv/data_struct.py

```diff
@@ -25,8 +26,73 @@ class ScheduleMode(enum.Enum):
     HOLD = 5
 
 
+class SettingsBitFieldData(BitFieldData):
+    LOCK_CONTROL = 7
+    VALVE_INSTALLED = 6
+    SLOW_REGULATION = 4
+    DISPLAY_FLIP = 3
+    VERTICAL_INSTALATION = 2
+    ADAPTABLE_REGULATION = 0
+
+    @property
+    def default_value(self):
+        return 0x22  # 2 Unknown bits maybe, but it failed if not set
+
+    @property
+    def lock_control(self):
+        return self._read_bit(SettingsBitFieldData.LOCK_CONTROL)
+
+    @lock_control.setter
+    def lock_control(self, value):
+        self._set_bit(SettingsBitFieldData.LOCK_CONTROL, value)
+
+    @property
+    def valve_installed(self):
+        return self._read_bit(SettingsBitFieldData.VALVE_INSTALLED)
+
+    @valve_installed.setter
+    def valve_installed(self, value):
+        self._set_bit(SettingsBitFieldData.VALVE_INSTALLED, value)
+
+    @property
+    def slow_regulation(self):
+        return self._read_bit(SettingsBitFieldData.SLOW_REGULATION)
+
+    @slow_regulation.setter
+    def slow_regulation(self, value):
+        self._set_bit(SettingsBitFieldData.SLOW_REGULATION, value)
+
+    @property
+    def display_flip(self):
+        return self._read_bit(SettingsBitFieldData.DISPLAY_FLIP)
+
+    @display_flip.setter
+    def display_flip(self, value):
+        self._set_bit(SettingsBitFieldData.DISPLAY_FLIP, value)
+
+    @property
+    def vertical_instalation(self):
+        return self._read_bit(SettingsBitFieldData.VERTICAL_INSTALATION)
+
+    @vertical_instalation.setter
+    def vertical_instalation(self, value):
+        self._set_bit(SettingsBitFieldData.VERTICAL_INSTALATION, value)
+
+    @property
+    def adaptable_regulation(self):
+        return self._read_bit(SettingsBitFieldData.ADAPTABLE_REGULATION)
+
+    @adaptable_regulation.setter
+    def adaptable_regulation(self, value):
+        self._set_bit(SettingsBitFieldData.ADAPTABLE_REGULATION, value)
+
+
 class SettingsData(eTRVData):
-    frost_protection_temperature = TemperatureField()
+    configuration_bits = BitField(
```

Comment:
I'm not a fan of such inner structures and I was trying to avoid them. What about using `eTRVField(name='config_bits')` and combine it together as a proper `BitField`? For example, definition can be:
```
class SettingsData(eTRVData):
    # ...
    vertical_instalation = BitField(name='config_bits', bit=SettingsBitFieldData.VERTICAL_INSTALATION)
    frost_protection_temperature = TemperatureField()
    # ...
```
Also, I would change the definition of these static values to `IntEnum`, and maybe move them to data_struct.py file, so all these definitions will be easily accessible for people who want to integrate this eTRV in other languages. I was trying to separate data structures from logic so it helps in analyzing how to communicate.

This changes are mostly refactor of your code, most of it is already done.